extensions/cornerstone/src/panels/PanelMeasurement.tsx

```diff
@@ -69,29 +85,33 @@ export default function PanelMeasurementTable({
       <div
         className="invisible-scrollbar overflow-y-auto overflow-x-hidden"
         ref={measurementsPanelRef}
-        data-cy={'trackedMeasurements-panel'}
+        data-cy={'measurements-panel'}
       >
-        <MeasurementTable
-          key="tracked"
-          title="Measurements"
-          data={measurements}
-          {...onArgs}
-          // onColor={changeColorMeasurement}
-        >
-          <MeasurementTable.Header>
-            {customHeader && (
-              <>
-                {typeof customHeader === 'function'
-                  ? customHeader({
-                      additionalFindings,
-                      measurements,
-                    })
-                  : customHeader}
-              </>
-            )}
-          </MeasurementTable.Header>
-          <MeasurementTable.Body />
-        </MeasurementTable>
+        {Array.from(measurements).map(([key, value]) => {
+          return (
+            <MeasurementTable
+              key={`${key}`}
+              title={title ? title : `Measurements for ${key}`}
+              data={value}
+              {...onArgs}
+              // onColor={changeColorMeasurement}
+            >
+              <MeasurementTable.Header>
+                {customHeader && (
+                  <>
+                    {typeof customHeader === 'function'
+                      ? customHeader({
+                          additionalFindings,
+                          measurements,
+                        })
+                      : customHeader}
+                  </>
+                )}
+              </MeasurementTable.Header>
+              <MeasurementTable.Body />
+            </MeasurementTable>
+          );
+        })}
         {additionalFindings.length > 0 && (
```

Comment:
The additional findings are a nested/recursive application of the grouping function - so that this is a sub-group of hte study grouping.